src/sync/app_credentials.cpp

```diff
@@ -161,6 +161,15 @@ AppCredentials AppCredentials::function(const bson::BsonDocument& payload)
                           });
 }
 
+AppCredentials AppCredentials::function(const std::string serialized_document)
```

Comment:
Could have this parameter be a const ref (`const std::string&`) since the lambda captures a copy of it anyways.

Reply:
Hm... I guess it could, but why are the other credentials methods taking `std::string`s directly? Is it an oversight or is it due to some subtle C++-ness that I don't understand?